File: english-hub-backend/src/main/java/com/example/englishhubbackend/service/FlashCardService.java
package com.example.englishhubbackend.service;

import com.example.englishhubbackend.dto.request.FlashCardCreateRequest;
import com.example.englishhubbackend.dto.request.FlashCardUpdateRequest;
import com.example.englishhubbackend.dto.response.FlashCardResponse;

import java.util.List;
import java.util.UUID;

public interface FlashCardService {
    List<FlashCardResponse> getAllFlashCardsFromDecks(UUID deckId);

    FlashCardResponse createFlashCard(UUID deckId, FlashCardCreateRequest flashCardCreateRequest);

    FlashCardResponse updateFlashCard(UUID flashCardId, FlashCardUpdateRequest flashCardUpdateRequest);

    void deleteFlashCard(UUID flashCardId);

    FlashCardResponse getFlashCardById(UUID deckId , UUID flashCardId);
}

File: english-hub-frontend/src/services/flashCardService.ts
import {
  FlashCardCreateRequest,
  FlashCardResponse,
  FlashCardUpdateRequest,
} from '@/types/flashCardType';
import axiosInstance from './axiosInstance';

export async function getAllFlashCards(
  deckId: string
): Promise<FlashCardResponse[]> {
  const response = await axiosInstance.get(`/decks/${deckId}/flashcards`);
  return response.data.result;
}
export async function getFlashCardById(
  deckId: string,
  flashCardId: string
): Promise<FlashCardResponse> {
  const response = await axiosInstance.get(
    `/decks/${deckId}/flashcards/${flashCardId}`
  );
  return response.data.result;
}

export async function createFlashCard(
  deckId: string,
  flashCardData: FlashCardCreateRequest
): Promise<FlashCardResponse> {
  const response = await axiosInstance.post(
    `/decks/${deckId}/flashcards`,
    flashCardData
  );
  return response.data.result;
}

export async function updateFlashCard(
  deckId: string,
  flashCardId: string,
  flashCardData: FlashCardUpdateRequest
): Promise<FlashCardResponse> {
  const response = await axiosInstance.put(
    `/decks/${deckId}/flashcards/${flashCardId}`,
    flashCardData
  );
  return response.data.result;
}

export async function deleteFlashCard(
  deckId: string,
  flashCardId: string
): Promise<string> {
  const response = await axiosInstance.delete(
    `/decks/${deckId}/flashcards/${flashCardId}`
  );
  return response.data.message;
}
